cassiopeia/plotting/local.py

```diff
@@ -0,0 +1,830 @@
+            _leaf_kwargs["text"].append(text)
+        else:
+            _node_kwargs["x"].append(x)
+            _node_kwargs["y"].append(y)
+            _node_kwargs["text"].append(text)
+    figure.add_trace(go.Scatter(**_leaf_kwargs))
+    figure.add_trace(go.Scatter(**_node_kwargs))
+
+    _leaf_colors = []
+    _node_colors = []
+    _leaf_kwargs.update({"x": [], "y": [], "text": []})
+    _node_kwargs.update({"x": [], "y": [], "text": []})
+    for node, color in node_colors.items():
+        x, y = node_coords[node]
+        text = f"<b>NODE</b><br>{node}"
+        if is_polar:
+            x, y = utilities.polar_to_cartesian(x, y)
+        if tree.is_leaf(node):
+            _leaf_kwargs["x"].append(x)
+            _leaf_kwargs["y"].append(y)
+            _leaf_kwargs["text"].append(text)
+            _leaf_colors.append(color)
+        else:
+            _node_kwargs["x"].append(x)
+            _node_kwargs["y"].append(y)
+            _node_kwargs["text"].append(text)
+            _node_colors.append(color)
+
+    _leaf_kwargs["marker_color"] = _leaf_colors
+    _node_kwargs["marker_color"] = _node_colors
+    figure.add_trace(go.Scatter(**_leaf_kwargs))
+    figure.add_trace(go.Scatter(**_node_kwargs))
+
+    # Plot all branches
+    _branch_kwargs = dict(
+        x=[],
+        y=[],
+        text=[],
+        line_color="black",
+        line_width=1,
+        mode="lines",
+        showlegend=False,
+        hoverinfo="text",
+    )
+    _branch_kwargs.update(branch_kwargs or {})
+    for branch, (xs, ys) in branch_coords.items():
+        if branch in branch_colors:
+            continue
+        _branch_kwargs["x"], _branch_kwargs["y"] = xs, ys
+        text = f"<b>BRANCH</b><br>{branch[0]}<br>{branch[1]}"
+        if is_polar:
+            (
+                _branch_kwargs["x"],
+                _branch_kwargs["y"],
+            ) = utilities.polars_to_cartesians(xs, ys)
+        _branch_kwargs["text"] = [text] * len(xs)
+        figure.add_trace(go.Scatter(**_branch_kwargs))
+
+    for branch, color in branch_colors.items():
+        xs, ys = branch_coords[branch]
+        _branch_kwargs["x"], _branch_kwargs["y"] = xs, ys
+        _branch_kwargs["line_color"] = color
+        text = f"<b>BRANCH</b><br>{branch[0]}<br>{branch[1]}"
+        if is_polar:
+            (
+                _branch_kwargs["x"],
+                _branch_kwargs["y"],
+            ) = utilities.polars_to_cartesians(xs, ys)
+        _branch_kwargs["text"] = [text] * len(xs)
+        figure.add_trace(go.Scatter(**_branch_kwargs))
+
+    # Colorstrips
+    _colorstrip_kwargs = dict(
+        x=[],
+        y=[],
+        text=[],
+        line_width=0,
+        fill="toself",
+        mode="lines",
+        showlegend=False,
+        hoverinfo="text",
+        hoveron="fills",
+    )
+    _colorstrip_kwargs.update(colorstrip_kwargs or {})
+    for colorstrip in colorstrips:
+        # Last element is text, but this can not be shown in static plotting.
```

Comment:
comment no longer necessary for this version it seems

Reply:
You're right. I copied over the matplotlib function as a template, so that's where these stragglers are coming from.